pkg/rancher-desktop/backend/wsl.ts

```diff
@@ -1713,4 +1719,29 @@ export default class WSLBackend extends events.EventEmitter implements VMBackend
     return super.rawListeners(event) as BackendEvents[eventName][];
   }
   // #endregion
+
+  protected async isAdminInstall(): Promise<boolean> {
```

Comment:
I think this would be clearer as a memoized getter:

```ts
protected #isAdminInstall: Promise<boolean> | undefined;

protected getIsAdminInstall(): Promise<boolean> {
  this.#isAdminInstall ??= new Promise((resolve) => {
    …
  });

  return this.#isAdminInstall;
}
```
And make all callers call `getIsAdminInstall()` directly (but it would only ever get triggered once).

Mostly, this clears up any worries about checking `this.adminInstall` before `isAdminInstall()` was invoked.